py/makeqstrdata.py

```diff
@@ -307,15 +344,17 @@ def compute_hash(qstr, bytes_hash):
 
 
 def qstr_escape(qst):
-    def esc_char(m):
-        c = ord(m.group(0))
+    def esc_char(c):
+        if re.match(r"^[a-zA-Z0-9_]$", c):
+            return c
+        c = ord(c)
         try:
             name = codepoint2name[c]
         except KeyError:
             name = "0x%02x" % c
         return "_" + name + "_"
 
-    return re.sub(r"[^A-Za-z0-9_]", esc_char, qst)
+    return "".join(map(esc_char, qst))
```

Comment:
Why doesn't this work in MicroPython, is it simply because `re.sub` is not always enabled on MicroPython?